ecs/src/nox/ecs/createMetaInformation.tpp

```diff
@@ -10,115 +101,146 @@ nox::ecs::createMetaInformation(const TypeIdentifier& typeIdentifier)
+nox::ecs::createMetaInformation(const TypeIdentifier& typeIdentifier,
                                 const std::vector<nox::event::Event::IdType>& interestingLogicEvents)
 {
+    using namespace create_meta_info_meta;
     static_assert(std::is_base_of<nox::ecs::Component, T>::value, "Type T must inherit from nox::ecs::Component");
-    MetaInformation metaInformation(typeIdentifier, sizeof(T));
-    
-    const auto construct = [](Component* component, const EntityId& id, EntityManager* manager)
-    { 
-        new(component)T(id, manager);
-    };
-
-    const auto destruct = [](Component* component)
-    {
-        static_cast<T*>(component)->~T();
-    };
-
-    const auto initialize = [](Component* component, const Json::Value& value)
-    {
-        static_cast<T*>(component)->initialize(value);
-    };
-
-    const auto awake = [](Component* component)
-    {
-        static_cast<T*>(component)->awake();
-    };
-
-    const auto activate = [](Component* component)
-    {
-        static_cast<T*>(component)->activate();
-    };
-
-    const auto deactivate = [](Component* component)
-    {
-        static_cast<T*>(component)->deactivate();   
-    };
-
-    const auto hibernate = [](Component* component)
-    {
-        static_cast<T*>(component)->hibernate();   
-    };
 
-    const auto update = [](Component* first, Component* last,
-                           const nox::Duration& deltaTime)
-    {
-        auto begin = static_cast<T*>(first);
-        auto end = static_cast<T*>(last);
-
-        while (begin != end)
-        {
-            begin->update(deltaTime);
-            ++begin;
-        }
-    };
+    MetaInformation info(typeIdentifier, sizeof(T));
 
-    const auto receiveLogicEvent = [](Component* first, Component* last,
-                                      const std::shared_ptr<nox::event::Event>& event)
+    info.construct = [](Component* component, const EntityId& id, EntityManager* manager)
     {
-        auto begin = static_cast<T*>(first);
-        auto end = static_cast<T*>(last);
-
-        while (begin != end)
-        {
-            begin->receiveLogicEvent(event);
-            ++begin;
-        }
+        new(component)T(id, manager);
     };
 
-    const auto receiveEntityEvent = [](Component* first,
-                                       Component* last,
-                                       const nox::ecs::Event& event)
+    info.destruct = [](Component* component)
     {
-        auto begin = static_cast<T*>(first);
-        auto end = static_cast<T*>(last);
-
-        while (begin != end)
-        {
-            begin->receiveEntityEvent(event);
-            ++begin;
-        }
+        static_cast<T*>(component)->~T();
     };
 
-    const auto moveConstruct = [](Component* dest, Component* src)
+    info.initialize =
+        getOperation<decltype(&Component::initialize),
+                     decltype(&T::initialize),
+                     decltype(info.initialize)>(
```

Comment:
I would prefer if this starting bracket was moved to the next line, so it becomes "([](Component* component),".